### PR TITLE
kernel: use tar.xz, git optional

### PR DESCRIPTION
**Pull request by Noxenious**
Per default use stable kernel by downloading tar.xz (fast download).
Optional use latest mainline kernel by git fetch. Must be enabled
in linux_git.bb by setting DEFAULT_PREFERENCE to "1".

Signed-off-by: Jens Lucius <info@jenslucius.com>

### CODE REVIEW

File: recipes-kernel/linux/linux_git.bb [hunk 1]
```diff
@@ -12,11 +12,17 @@ require linux.inc
...
+# If you want to use latest git version set to "1"
+DEFAULT_PREFERENCE = "-1" 
+
 KERNEL_EXTRA_ARGS += "LOADADDR=${UBOOT_ENTRYPOINT}"
 	
-PV = "4.1.2"
-SRCREV_pn-${PN} = "5cf9896dc5c72a6c68c36140568b755f697f7760"
-SRC_URI += "git://git.kernel.org/pub/scm/linux/kernel/git/stable/linux-stable.git;protocol=git;branch=linux-4.1.y \
+# 4.2 rc4
+PV = "4.2"
```
Comment (naguirre) [inline recipes-kernel/linux/linux_git.bb]: should be 4.2rc4  ?

**Comment by Noxenious** [comment]
If I try to set PV to 4.2rc4 or 4.2.rc4 I get this error when building. If you know how to set it, let me know and I correct it.

 ValueError: invalid literal for int() with base 10: 'rc4'

**Comment by naguirre** [comment]
i think that this PV should work : "4.2+rc4.git${SRCPV}"



-- 
Nicolas Aguirre
Mail: aguirre.nicolas@gmail.com
Web: http://www.calaos.fr
Blog: http://dev.enlightenment.fr/~captainigloo/

**Comment by ebutera** [comment]
if i'm not wrong for rc kernels i've seen something like:

"4.1+4.2...."

i think it was used to keep correct versioning (4.2+rc could happen to be > 4.2)

**Comment by naguirre** [comment]
ah yes you're right

**Comment by Noxenious** [comment]
Updated

**Comment by ebutera** [comment]
for me it's ok, but for the record:
https://www.mail-archive.com/openembedded-core@lists.openembedded.org/msg67604.html

i don't know if it means that even a "4.2+git...." is not ok because of the period between 4 and 2...

**Comment by Noxenious** [comment]
Can this be merged or shall I change it? I want to add a fix for the U-Boot machines.

**Pull request merged**